### PR TITLE
Raise an exception if the block arity is nonzero and isn't same with capture parameter length

### PR DESCRIPTION
**Pull request by namusyaka**
ref #1868
As @ujifgc said, I stopped adjusting captures at runtime.
Instead, an exception will be raised if the arity is wrong.

Please push commits into this branch if you have more friendly error message.

### CODE REVIEW

File: padrino-core/lib/padrino-core/application/routing.rb [hunk 1]
```diff
@@ -21,6 +21,10 @@ module Routing
     # Raised when a route was invalid or cannot be processed.
     class UnrecognizedException < RuntimeError; end
 
+    # Raised when block arity was nonzero and was not same with
+    # captured parameter length.
+    class WrongArityException < RuntimeError; end
```
Comment (skade) [inline padrino-core/lib/padrino-core/application/routing.rb]: I would make this a subclass of ArgumentError, which is a `StandardError`, though.

In general, wouldn't `StandardError` be the better pick anyways?

Comment (ujifgc) [inline padrino-core/lib/padrino-core/application/routing.rb]: It should be definitely ArgumentError, not Runtime one.

```
    class RouteBlockArityError < ArgumentError
      def initialize(path, block_arity, required_arity)
        super "route block arity does not match path '#{path}' (#{block_arity} for #{required_arity})"
      end
    end
```

File: padrino-core/lib/padrino-core/application/routing.rb [hunk 2]
```diff
@@ -539,6 +542,12 @@ def route(verb, path, *args, &block)
 
         invoke_hook(:padrino_route_added, route, verb, path, args, options, block)
 
+        block_parameter_length = route.block_parameter_length
+        if block_arity.nonzero? && block_parameter_length != block_arity
```
Comment (ujifgc) [inline padrino-core/lib/padrino-core/application/routing.rb]: It should allow negative arity values that are not equal to the number of captures. ref: (http://ruby-doc.org/core-1.9.3/Proc.html#method-i-arity)[negative arity]

**Comment by namusyaka** [comment]
Thank you for great reviewing and fixing!

**Pull request merged**